Device objects in User.devices have different fields on iOS and Android SDKs

## What did you do?
On iOS app tried to load channels one of which in `members` field have a user created on Android app

## What did you expect to happen?
Channels presented correctly

## What happened instead?
Decoding error

## GetStream Environment
**GetStream Chat version: 2.2.0**
**iOS version: 13.4**
**Swift version: Swift 5**
**Xcode version: 11.4.1**
**Device: iPhone 11 Pro Simulator**

## Additional context
`Device` struct in StreamChatClient has required fields `id` and `created_at`, where `Device` class in 
stream-chat-android-client has `id` and `push_provider`, which leads to decoding error on iOS

For old users I still get JSON without `created_at` field:
`"devices" : [
                {
                  "id" : "some_id",
                  "push_provider" : "firebase"
                }
              ],`
and therefore a decoding error
`[ERROR] ❌  keyNotFound(CodingKeys(stringValue: "created_at", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "channels", intValue: nil), _JSONKey(stringValue: "Index 12", intValue: 12), CodingKeys(stringValue: "messages", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "mentioned_users", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "devices", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"created_at\", intValue: nil) (\"created_at\").", underlyingError: nil)) in parse(data:response:error:completion:)[309]`

Should it be checked for new Android users?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### 🔄 Changed
 
+### 🐞 Fixed
+- Device object sometimes failing to decode on API calls [#266](https://github.com/GetStream/stream-chat-swift/issues/266)
+
 # [2.2.6](https://github.com/GetStream/stream-chat-swift/releases/tag/2.2.6)
 _July 07, 2020_
 

--- a/Sources/Client/Model/Device.swift
+++ b/Sources/Client/Model/Device.swift
@@ -17,7 +17,7 @@ public struct Device: Codable, Equatable {
     /// A device identifier.
     public let id: String
     /// A created date.
-    public let created: Date
+    public let created: Date?
     
     /// Init a device for Push Notifications.
     ///

